internal/fileio/fileio.go

```diff
@@ -39,13 +41,37 @@ func ParseHalConfig(halPath string) (*config.Hal, error) {
 		return nil, fmt.Errorf("unable to read %q. Error: %v", halPath, err)
 	}
 
+	return ParseHalConfigData(data)
+}
+
+// ParseHalConfigs reads the YAML files at halPaths parses it into a *config.Hal.
+// The config.Hal is validated; if there are any errors, they will be returned
+// in error and *config.Hal will be nil.
+func ParseHalConfigs(paths []string) (*config.Hal, error) {
+	merged, err := mergeConfigs(paths)
+	if err != nil {
+		return nil, err
+	}
+
+	data, err := yaml.Marshal(merged)
```

Comment:
I wonder if there's a way to do this without having to round-trip to/from YAML.

Reply:
Should we leave a TODO to explore alternatives?

Reply:
It looks like there's a [Merge](https://pkg.go.dev/google.golang.org/protobuf/proto?tab=doc#Merge) function generated for types `proto.Message` that we would be able to use here.   (Per the godoc for that function, you can also just repeatedly unmarshal into the same proto specifying the `Merge` strategy.)

I think that would be preferable as it uses a well-documented upstream merge strategy that uses the type information we have on our messages.